drafts/web-editor-monaco-with-language-server-support.md

```diff
@@ -0,0 +1,971 @@
+---
+title: Create a custom web editor using Monaco and Language Server Protocol (LSP)
+domain: software-engineering-corner.hashnode.dev
+tags: Web Editor, Monaco, Monaco Language Client, Language Server, LSP
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1719718605895/bv2LfyFWS.jpg?auto=format
+publishAs: cjayashantha, NipunaMarcusZuhlke
+hideFromHashnodeCommunity: false
+--- 
+
+# Create a custom web editor using Monaco and Language Server Protocol (LSP)
+
+If you are developing a domain-specific language or custom language, chances are high that current editors might not support features such as syntax highlighting, diagnostics, and code autocompletion. To enable these features, you need a Language Server Protocol (LSP) and sometimes a custom editor. In this article, we will discuss how to develop a web editor and connect it to a language server using the WebSocket protocol.
+
+### What is Monaco Editor?
+Monaco Editor is the code editor that powers Visual Studio Code, known for its performance, rich API, and extensive feature set. It offers:
+
+- Syntax highlighting 
+- IntelliSense (auto-completion)
+- Code navigation (go to definition, find references)
+- Multiple language support
+
+### What is Language Server Protocol (LSP) ?
+LSP is a protocol used to provide language-specific features in a language-agnostic way. It decouples the editor from the language-specific logic, allowing you to support various languages with minimal effort. LSP offers:
+
+- Syntax checking
+- Auto-completions 
+- Hover information
+- Code formatting
+- Refactorings
+
+# Creating Web Editor
+
+Let's start developing a web editor for a custom language server.
+
+For this example, we will develop our own mock language server fronted with websocket. Full implementation available at below location
+
+https://github.com/NipunaMarcus/hellols/tree/websocket-launcher
```

Comment:
You could use markdown syntax for links:
`[link text](https://link-url-here.org)`

This is then displayed as [link text](https://link-url-here.org)